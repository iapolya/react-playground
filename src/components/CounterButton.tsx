import React, { Component } from 'react';

interface IButtonProps {
  color?: string;
  count?: number;
}

interface IButtonState extends IButtonProps {
  count: number
}

class CounterButton extends Component<IButtonProps, IButtonState> {
  constructor(props: IButtonProps) {
    super(props);
    this.state = { count: 1 };
  }

  shouldComponentUpdate(nextProps: IButtonProps, nextState: IButtonState) {
    return this.state.count !== nextState.count;
  }

  render() {
    console.log('counter');
    return (
      <button
        color={this.props.color}
        onClick={() => this.setState((state) => ({ count: state.count + 1 }))}>
        Count: {this.state.count}
      </button>
    );
  }
}

export default CounterButton;
