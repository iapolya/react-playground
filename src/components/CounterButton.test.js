import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

it('expect to render card button component', () => {
  expect(shallow(<CounterButton />)).toMatchSnapshot();
});

it('increment counter', () => {
  const wrapper = shallow(<CounterButton />);
  wrapper.find('button').simulate('click');
  expect(wrapper.state()).toEqual({ count: 2 });
});
