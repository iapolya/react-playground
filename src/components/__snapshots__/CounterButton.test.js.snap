// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card button component 1`] = `
<button
  onClick={[Function]}
>
  Count: 
  1
</button>
`;
